src/utils.ts

```diff
@@ -1,10 +1,62 @@
-export const isEnvKeyValid = (request: any) => request && request.envKey && typeof request.envKey === 'string';
+import { EnvService, EnvServiceMethod, EnvServiceMethods } from './api/Env';
+import { EnvRegistryItem } from './api/EnvRegistry';
 
-export const isEnvValid = (request: any) => {
-  return request && request.env && request.env.accessPoints && request.env.accessPoints[0];
+export const isRegisterRequestValid = (request: any) => {
+  const props = ['envKey', 'env'];
+  return !(
+    !request ||
+    Object.entries(request).length !== 2 ||
+    !props.every((p) => Object.getOwnPropertyNames(request).includes(p))
+  );
 };
 
+export const isEnvKeyValid = (request: EnvRegistryItem) => typeof request.envKey === 'string';
+
+export const isEnvValid = (request: EnvRegistryItem) =>
+  request.env &&
+  request.env.constructor === Object &&
+  Object.entries(request.env).length === 1 &&
+  request.env.services &&
+  isEnvServiceValid(request.env.services);
+
+const isEnvServiceValid = (services: EnvService[]) =>
+  services.every(
+    (service) =>
+      service.constructor === Object &&
+      Object.entries(service).length === 3 &&
+      !(!service.serviceName || !isEnvServiceNameValid(service.serviceName)) &&
+      !(!service.url || !isEnvUrlValid(service.url)) &&
+      !(!service.methods || !isEnvMethodsValid(service.methods))
+  );
+
+const isEnvServiceNameValid = (serviceName: string) => serviceName && typeof serviceName === 'string';
+const isEnvUrlValid = (url: string) => url && typeof url === 'string';
+const isEnvMethodsValid = (methods: EnvServiceMethods) => {
+  let isValid = true;
+  if (methods.constructor !== Object) {
```

Comment:
let isValid = methods.constructor === Object;
if (isValid) {
  isValid  = Object.**values**(methods).**every/some**((methodValue) =>  isEnvMethodValid(methodValue)) 
}
// check this : methodValue && methodValue.constructor === Object in isEnvMethodValid